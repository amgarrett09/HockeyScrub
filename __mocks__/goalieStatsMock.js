const stats = [
  {
    season: '20182019',
    stat: {
      timeOnIce: '1658:22',
      ot: 1,
      shutouts: 0,
      ties: 0,
      wins: 16,
      losses: 12,
      saves: 871,
      powerPlaySaves: 83,
      shortHandedSaves: 30,
      evenSaves: 758,
      shortHandedShots: 31,
      evenShots: 825,
      powerPlayShots: 92,
      savePercentage: 0.919,
      goalAgainstAverage: 2.79,
      games: 30,
      gamesStarted: 29,
      shotsAgainst: 948,
      goalsAgainst: 77,
      powerPlaySavePercentage: 90.21739130434783,
      shortHandedSavePercentage: 96.7741935483871,
      evenStrengthSavePercentage: 91.87878787878788,
    },
    team: {
      id: 4,
      name: 'Philadelphia Flyers',
      link: '/api/v1/teams/4',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
];

const career = {
  timeOnIce: '1658:22',
  ot: 1,
  shutouts: 0,
  wins: 16,
  losses: 12,
  saves: 871,
  powerPlaySaves: 83,
  shortHandedSaves: 30,
  evenSaves: 758,
  shortHandedShots: 31,
  evenShots: 825,
  powerPlayShots: 92,
  savePercentage: 0.918776,
  goalAgainstAverage: 2.78587,
  games: 30,
  gamesStarted: 29,
  shotsAgainst: 948,
  goalsAgainst: 77,
  timeOnIcePerGame: '55:16',
  powerPlaySavePercentage: 90.21739130434783,
  shortHandedSavePercentage: 96.7741935483871,
  evenStrengthSavePercentage: 91.87878787878788,
};

module.exports = {
  stats,
  career,
};
