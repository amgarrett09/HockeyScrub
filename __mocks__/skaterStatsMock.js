const stats = [
  {
    season: '20162017',
    stat: {
      timeOnIce: '63:45',
      assists: 0,
      goals: 0,
      pim: 0,
      shots: 3,
      games: 7,
      hits: 12,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '00:22',
      evenTimeOnIce: '63:23',
      penaltyMinutes: '0',
      faceOffPct: 50.0,
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: -1,
      points: 0,
      shifts: 90,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
  {
    season: '20162017',
    stat: {
      assists: 25,
      goals: 12,
      pim: 37,
      games: 48,
      penaltyMinutes: '37',
      plusMinus: 15,
      points: 37,
    },
    team: {
      name: 'Grand Rapids Griffins',
      link: '/api/v1/teams/null',
    },
    league: {
      name: 'AHL',
      link: '/api/v1/league/null',
    },
    sequenceNumber: 978,
  },
  {
    season: '20172018',
    stat: {
      timeOnIce: '691:12',
      assists: 17,
      goals: 7,
      pim: 39,
      shots: 83,
      games: 48,
      hits: 53,
      powerPlayGoals: 1,
      powerPlayPoints: 2,
      powerPlayTimeOnIce: '68:25',
      evenTimeOnIce: '617:34',
      penaltyMinutes: '39',
      faceOffPct: 12.5,
      shotPct: 8.4,
      gameWinningGoals: 1,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '05:13',
      blocked: 38,
      plusMinus: -7,
      points: 24,
      shifts: 901,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
  {
    season: '20172018',
    stat: {
      assists: 7,
      goals: 7,
      pim: 34,
      games: 16,
      penaltyMinutes: '34',
      plusMinus: -5,
      points: 14,
    },
    team: {
      name: 'Grand Rapids Griffins',
      link: '/api/v1/teams/null',
    },
    league: {
      name: 'AHL',
      link: '/api/v1/league/null',
    },
    sequenceNumber: 978,
  },
  {
    season: '20182019',
    stat: {
      timeOnIce: '1184:04',
      assists: 26,
      goals: 21,
      pim: 36,
      shots: 130,
      games: 72,
      hits: 74,
      powerPlayGoals: 3,
      powerPlayPoints: 6,
      powerPlayTimeOnIce: '121:37',
      evenTimeOnIce: '1061:16',
      penaltyMinutes: '36',
      faceOffPct: 33.33,
      shotPct: 16.2,
      gameWinningGoals: 4,
      overTimeGoals: 1,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '01:11',
      blocked: 38,
      plusMinus: 12,
      points: 47,
      shifts: 1525,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    league: {
      id: 133,
      name: 'National Hockey League',
      link: '/api/v1/league/133',
    },
    sequenceNumber: 1,
  },
];

const career = {
  timeOnIce: '1939:01',
  assists: 43,
  goals: 28,
  pim: 75,
  shots: 216,
  games: 127,
  hits: 139,
  powerPlayGoals: 4,
  powerPlayPoints: 8,
  powerPlayTimeOnIce: '190:24',
  evenTimeOnIce: '1742:13',
  penaltyMinutes: '75',
  faceOffPct: 30.0,
  shotPct: 13.0,
  gameWinningGoals: 5,
  overTimeGoals: 1,
  shortHandedGoals: 0,
  shortHandedPoints: 0,
  shortHandedTimeOnIce: '06:24',
  blocked: 76,
  plusMinus: 4,
  points: 71,
  shifts: 2516,
  timeOnIcePerGame: '15:16',
  evenTimeOnIcePerGame: '13:43',
  shortHandedTimeOnIcePerGame: '00:03',
  powerPlayTimeOnIcePerGame: '01:29',
};

const lastFive = [
  {
    season: '20182019',
    stat: {
      timeOnIce: '20:25',
      assists: 0,
      goals: 0,
      pim: 4,
      shots: 4,
      games: 1,
      hits: 1,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '02:40',
      evenTimeOnIce: '17:45',
      penaltyMinutes: '4',
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: -1,
      points: 0,
      shifts: 24,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 5,
      name: 'Pittsburgh Penguins',
      link: '/api/v1/teams/5',
    },
    date: '2019-04-04',
    isHome: false,
    isWin: false,
    isOT: false,
    game: {
      gamePk: 2018021245,
      link: '/api/v1/game/2018021245/feed/live',
      content: {
        link: '/api/v1/game/2018021245/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '19:26',
      assists: 1,
      goals: 2,
      pim: 0,
      shots: 5,
      games: 1,
      hits: 0,
      powerPlayGoals: 1,
      powerPlayPoints: 1,
      powerPlayTimeOnIce: '00:22',
      evenTimeOnIce: '18:48',
      penaltyMinutes: '0',
      shotPct: 40.0,
      gameWinningGoals: 1,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:16',
      blocked: 0,
      plusMinus: 2,
      points: 3,
      shifts: 27,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 5,
      name: 'Pittsburgh Penguins',
      link: '/api/v1/teams/5',
    },
    date: '2019-04-02',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021233,
      link: '/api/v1/game/2018021233/feed/live',
      content: {
        link: '/api/v1/game/2018021233/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:18',
      assists: 3,
      goals: 0,
      pim: 0,
      shots: 1,
      games: 1,
      hits: 0,
      powerPlayGoals: 0,
      powerPlayPoints: 2,
      powerPlayTimeOnIce: '04:20',
      evenTimeOnIce: '11:51',
      penaltyMinutes: '0',
      shotPct: 0.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:07',
      blocked: 1,
      plusMinus: 1,
      points: 3,
      shifts: 22,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 6,
      name: 'Boston Bruins',
      link: '/api/v1/teams/6',
    },
    date: '2019-03-31',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021219,
      link: '/api/v1/game/2018021219/feed/live',
      content: {
        link: '/api/v1/game/2018021219/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:19',
      assists: 2,
      goals: 1,
      pim: 0,
      shots: 4,
      games: 1,
      hits: 2,
      powerPlayGoals: 0,
      powerPlayPoints: 0,
      powerPlayTimeOnIce: '01:00',
      evenTimeOnIce: '15:19',
      penaltyMinutes: '0',
      shotPct: 25.0,
      gameWinningGoals: 0,
      overTimeGoals: 0,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: 3,
      points: 3,
      shifts: 20,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 1,
      name: 'New Jersey Devils',
      link: '/api/v1/teams/1',
    },
    date: '2019-03-29',
    isHome: true,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021199,
      link: '/api/v1/game/2018021199/feed/live',
      content: {
        link: '/api/v1/game/2018021199/content',
      },
    },
  }, {
    season: '20182019',
    stat: {
      timeOnIce: '16:58',
      assists: 1,
      goals: 2,
      pim: 0,
      shots: 4,
      games: 1,
      hits: 0,
      powerPlayGoals: 1,
      powerPlayPoints: 1,
      powerPlayTimeOnIce: '02:46',
      evenTimeOnIce: '14:12',
      penaltyMinutes: '0',
      shotPct: 50.0,
      gameWinningGoals: 1,
      overTimeGoals: 1,
      shortHandedGoals: 0,
      shortHandedPoints: 0,
      shortHandedTimeOnIce: '00:00',
      blocked: 0,
      plusMinus: 1,
      points: 3,
      shifts: 21,
    },
    team: {
      id: 17,
      name: 'Detroit Red Wings',
      link: '/api/v1/teams/17',
    },
    opponent: {
      id: 7,
      name: 'Buffalo Sabres',
      link: '/api/v1/teams/7',
    },
    date: '2019-03-28',
    isHome: false,
    isWin: true,
    isOT: false,
    game: {
      gamePk: 2018021189,
      link: '/api/v1/game/2018021189/feed/live',
      content: {
        link: '/api/v1/game/2018021189/content',
      },
    },
  },
];

module.exports = {
  stats,
  career,
  lastFive,
};
