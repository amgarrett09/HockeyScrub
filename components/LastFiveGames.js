import React from 'react';
import PropTypes from 'prop-types';

const LastFiveGames = ({ position, games }) => {
  const skaterTable = (
    <div className="box">
      <h2 className="title is-2">Last Five Games</h2>
      <table className="table is-fullwidth is-narrow">
        <thead>
          <tr>
            <th>Opponent</th>
            <th><abbr title="Goals">G</abbr></th>
            <th><abbr title="Assists">A</abbr></th>
            <th><abbr title="Points">P</abbr></th>
            <th><abbr title="Plus / Minus">+/-</abbr></th>
            <th><abbr title="Penalty in Minutes">PIM</abbr></th>
            <th><abbr title="Powerplay Goals">PPG</abbr></th>
            <th><abbr title="Powerplay Points">PPP</abbr></th>
            <th><abbr title="Shorthanded Goals">SHG</abbr></th>
            <th><abbr title="Shorthanded Points">SHP</abbr></th>
            <th><abbr title="Game Winning Goals">GWG</abbr></th>
            <th><abbr title="Overtime Goals">OTG</abbr></th>
            <th><abbr title="Shots">S</abbr></th>
            <th><abbr title="Minutes">MIN</abbr></th>
          </tr>
        </thead>

        <tbody>
          {games.map((game, i) => (
            <tr key={game.opponent.name + i.toString()}>
              <td>{game.opponent.name}</td>
              <td>{game.stat.goals}</td>
              <td>{game.stat.assists}</td>
              <td>{game.stat.points}</td>
              <td>{game.stat.plusMinus}</td>
              <td>{game.stat.penaltyMinutes}</td>
              <td>{game.stat.powerPlayGoals}</td>
              <td>{game.stat.powerPlayPoints}</td>
              <td>{game.stat.shortHandedGoals}</td>
              <td>{game.stat.shortHandedPoints}</td>
              <td>{game.stat.gameWinningGoals}</td>
              <td>{game.stat.overTimeGoals}</td>
              <td>{game.stat.shots}</td>
              <td>{game.stat.timeOnIce}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );

  const goalieTable = (
    <div className="box">
      <h2 className="title is-2">Last Five Games</h2>
      <table className="table is-fullwidth">
        <thead>
          <tr>
            <th>Opponent</th>
            <th><abbr title="Shots Against">SA</abbr></th>
            <th><abbr title="Goals Against">GA</abbr></th>
            <th><abbr title="Saves">S</abbr></th>
            <th><abbr title="Save Percentage">S%</abbr></th>
            <th><abbr title="Minutes">MIN</abbr></th>
          </tr>
        </thead>

        <tbody>
          {games.map((game, i) => (
            <tr key={game.opponent.name + i.toString()}>
              <td>{game.opponent.name}</td>
              <td>{game.stat.shotsAgainst}</td>
              <td>{game.stat.goalsAgainst}</td>
              <td>{game.stat.saves}</td>
              <td>{game.stat.savePercentage}</td>
              <td>{game.stat.timeOnIce}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );

  const isGoalie = position === 'Goalie';

  return (
    <React.Fragment>
      {!isGoalie && skaterTable}
      {isGoalie && goalieTable}
    </React.Fragment>
  );
};

LastFiveGames.propTypes = {
  position: PropTypes.string.isRequired,
  games: PropTypes.arrayOf(PropTypes.shape({
    season: PropTypes.string,
  })).isRequired,
};

export default LastFiveGames;
