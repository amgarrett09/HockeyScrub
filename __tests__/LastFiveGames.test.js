/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import LastFiveGames from '../components/LastFiveGames';
import { lastFive } from '../__mocks__/skaterStatsMock';

describe('LastFiveGames', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <LastFiveGames position="Left Wing" games={lastFive} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
