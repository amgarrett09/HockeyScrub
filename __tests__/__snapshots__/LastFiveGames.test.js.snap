// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastFiveGames should match snapshot 1`] = `
<div
  className="stat-card"
>
  <h2>
    Last Five Games
  </h2>
  <table>
    <thead>
      <tr>
        <th>
          Opponent
        </th>
        <th>
          G
        </th>
        <th>
          A
        </th>
        <th>
          P
        </th>
        <th>
          +/-
        </th>
        <th>
          PIM
        </th>
        <th>
          PPG
        </th>
        <th>
          PPP
        </th>
        <th>
          SHG
        </th>
        <th>
          SHP
        </th>
        <th>
          GWG
        </th>
        <th>
          OTG
        </th>
        <th>
          S
        </th>
        <th>
          MIN
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Pittsburgh Penguins
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          -1
        </td>
        <td>
          4
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          4
        </td>
        <td>
          20:25
        </td>
      </tr>
      <tr>
        <td>
          Pittsburgh Penguins
        </td>
        <td>
          2
        </td>
        <td>
          1
        </td>
        <td>
          3
        </td>
        <td>
          2
        </td>
        <td>
          0
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          1
        </td>
        <td>
          0
        </td>
        <td>
          5
        </td>
        <td>
          19:26
        </td>
      </tr>
      <tr>
        <td>
          Boston Bruins
        </td>
        <td>
          0
        </td>
        <td>
          3
        </td>
        <td>
          3
        </td>
        <td>
          1
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          2
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          1
        </td>
        <td>
          16:18
        </td>
      </tr>
      <tr>
        <td>
          New Jersey Devils
        </td>
        <td>
          1
        </td>
        <td>
          2
        </td>
        <td>
          3
        </td>
        <td>
          3
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          4
        </td>
        <td>
          16:19
        </td>
      </tr>
      <tr>
        <td>
          Buffalo Sabres
        </td>
        <td>
          2
        </td>
        <td>
          1
        </td>
        <td>
          3
        </td>
        <td>
          1
        </td>
        <td>
          0
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
        <td>
          4
        </td>
        <td>
          16:58
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
